packages/core/src/extension/extension-client.ts

```diff
@@ -0,0 +1,54 @@
+export interface InitializeExtensionServiceOptions {
+    extensionId: string;
+    parentOrigin: string;
+}
+
+export enum ExtensionListenEventType {
+    CheckoutLoaded = 'CHECKOUT_LOADED',
+    ShippingCountryChange = 'SHIPPING_COUNTRY_CHANGE',
+}
+
+export interface CheckoutLoadedEvent {
+    type: ExtensionListenEventType.CheckoutLoaded;
+}
+
+export interface ShippingCountryChangeEvent {
+    type: ExtensionListenEventType.CheckoutLoaded;
+    payload: {
+        countryCode: string;
+    };
+}
+
+export interface ExtensionListenEventMap {
+    [ExtensionListenEventType.CheckoutLoaded]: CheckoutLoadedEvent;
+    [ExtensionListenEventType.ShippingCountryChange]: ShippingCountryChangeEvent;
+}
+
+export enum ExtensionCommandType {
+    FRAME_LOADED = 'FRAME_LOADED',
+    RELOAD_CHECKOUT = 'RELOAD_CHECKOUT',
+    SHOW_LOADING_INDICATOR = 'SHOW_LOADING_INDICATOR',
+}
```

Comment:
Do we still have `export const enum ExtensionCommandType` for `checkout-js`?

Reply:
I think so at least for now, unless we converge one section 🤔 Can combine that as a last stage fast follow I think.